src/main/java/com/marklogic/kafka/connect/source/ConstraintColumnHandler.java

```diff
@@ -0,0 +1,11 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.client.datamovement.RowBatcher;
+
+import java.util.Map;
+
+public interface ConstraintColumnHandler {
+    String injectConstraintIntoQuery(Map<String, Object> parsedConfig, String constraintColumn, String constraintValue);
+    String getMaxConstraintValue(String currentQuery, long queryStartTimeInMillis, String constraintColumn);
+    void addQueryToRowBatcher(String currentQuery, RowBatcher<?> rowBatcher);
```

Comment:
This 3rd method a really nice addition, and I think it means the following things:

1. The CCH implementation can be given the currentQuery and can modify it AND can hold onto it as a private field
2. injectConstraintIntoQuery doesn't need to be exposed in this interface
3. getMaxConstraintValue doesn't need currentQuery passed to it because the CCH already knows what it is
4. (You're not going to like this part) I don't think we need a QueryContext anymore :) . That's because if CCH is holding onto the currentQuery, it doesn't need to go into QueryContext, which means we don't need QueryContext anymore, we only need a RowBatcher and a RowBatcherBuilder. 

I think that means that what CCH is really about is managing the user's query... though it can't be called QueryManager since the Java Client already uses that term. But based on your PR, it fulfills two needs for the Task class:

1. It takes the user's query and makes the correct call to withBatchView on the RowBatcher (including support for the constraintColumn)
2. If the user specified a constraint column, the Task class will call getMaxConstraintValue

I think `QueryHandler` is what we want - it basically subsumes QueryContext and ConstraintColumnHandler. Ping me for more chat about this, but I think that's where the solution is evolving towards:

```
interface QueryHandler
  String addQueryToRowBatcher(String userQuery, RowBatcher)
  String getMaxConstraintValue(long queryStartTimeInMillis, String constraintColumnName)
```

Note that I think we can safely use the name "userQuery" in the interface, though the QH impl will likely call this "currentQuery" in its implementation since it may have been modified. And I think `addQuery` can return the query to make it easier to test this.